pages/index.tsx

```diff
@@ -7,14 +7,27 @@ import { Button } from '@/components/ui/button'
 import { useCurrentUser } from '@/hooks/query/user'
 
 const HomePage: NextPage = () => {
-  const { user } = useCurrentUser()
+  const { user, error, isLoading } = useCurrentUser()
   const router = useRouter()
+  const [errorMessage, setErrorMessage] = useState<string | null>(null)
 
   useEffect(() => {
-    if (!user) {
+    if (error) {
+      setErrorMessage('An error occurred while fetching user data. Please try again later.')
+    }
+
+    if (user && !isLoading) {
+      router.replace('/dashboard')
+    }
+
+    if (!user && !isLoading) {
```

Comment:
Hi @HarshMN2345
Thanks for contributing 😄.

This is the homepage as we would like a Landing page here. We don't want to redirect the user to `/dashboard`.

Also, the dashboard is at `app.localhost:3000/dashboard` subdomain 😄

But great code 🎉